### PR TITLE
Applets/SwKbd: Software Keyboard workaround

### LINKED ISSUE #3477: Can we just use System Username instead of hardcoded string?
Basically, couldn't you just use the System Username set in configuration instead of `Citra` that is hardcoded in https://github.com/citra-emu/citra/blob/master/src/core/hle/applets/swkbd.cpp#L80 ?

You know, until the new software keyboard is ready?

### PR DESCRIPTION
**Pull request by adityaruplaha**
Fixes #3477
PR may be reviewed commit by commit.
I haven't bothered to maintain clang-format as #3410 hasn't been merged yet. 

I won't do the UI as its only a temporary workaround. Merge only if the proper implementation is not done by end of April.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3488)

### CODE REVIEW

File: src/citra/config.cpp [hunk 1]
```diff
@@ -94,6 +94,9 @@ void Config::ReadValues() {
     Settings::values.use_frame_limit = sdl2_config->GetBoolean("Renderer", "use_frame_limit", true);
     Settings::values.frame_limit =
         static_cast<u16>(sdl2_config->GetInteger("Renderer", "frame_limit", 100));
+    
+    Settings::values.touch_device =
+        sdl2_config->Get("Renderer", "swkbd_input", "Citra");
```
Comment (wwylele) [inline src/citra/config.cpp]: Why is this in Renderer?

Comment (adityaruplaha) [reply src/citra/config.cpp]: Where should I put it?

**Comment by valentinvanelslande** [comment]
I just implemented it 9 hours ago in my 173 build
My implementation is millions of times better than yours

**Comment by adityaruplaha** [comment]
Just open a PR then. As soon as you open one, I'll close this.

File: src/core/hle/applets/swkbd.cpp [hunk 2]
```diff
@@ -76,8 +77,9 @@ ResultCode SoftwareKeyboard::StartImpl(Service::APT::AppletStartupParameter cons
 void SoftwareKeyboard::Update() {
     // TODO(Subv): Handle input using the touch events from the HID module
 
-    // TODO(Subv): Remove this hardcoded text
-    std::u16string text = Common::UTF8ToUTF16("Citra");
+    // TODO(Subv): Remove this hardcoded text <- Done!
```
Comment (FearlessTobi) [inline src/core/hle/applets/swkbd.cpp]: I don't think this comment should be kept, if it's not valid anymore

**Comment by adityaruplaha** [comment]
@valentinvanelslande
Well, I don't see the point of using your implementation. I've made this minimal coz the real thing is being worked upon now. But I'll admit its way way better & cleaner.

File: src/core/hle/applets/swkbd.cpp (hunk 2 above)
Comment (adityaruplaha) [reply src/core/hle/applets/swkbd.cpp]: OK.

File: src/citra/config.cpp (hunk 1 above)
Comment (wwylele) [reply src/citra/config.cpp]: Feel free to make a new group if you don't find any group it belongs to

**Comment by valentinvanelslande** [comment]
My implementation video:
[video.zip](https://github.com/citra-emu/citra/files/1793401/video.zip)
It seems there is a limit of swkbds...

**Comment by adityaruplaha** [comment]
@valentinvanelslande See this post.
https://community.citra-emu.org/t/software-keyboard-not-implemented-workaround/132/8

**Comment by valentinvanelslande** [comment]
I did exactly the same as
Game asks for software keyboard
Citra pauses emulation
A new Qt dialog box opens, asking user to enter text.
User enters : “abcd123Q”
Bingo ! Program resumes emulation, sends “abcd123Q” to the game, which parses it.

But my implementation opens citra-kbd/citra-kbd.exe, the keyboard writes the text to a file, calls exit(button id), Citra reads the file, deletes it, converts the system() result to SwkbdResult and sends the text to the game.
I already tried with a callback, but the game froze.
My implementation fixes a lot of games and yours none.

**Comment by adityaruplaha** [comment]
None? What do you mean? It'll allow the user the set the text before the game requests it. Your implementation has far more machine cycles. So it's not worth it.

**Comment by adityaruplaha** [comment]
TODO: Move setting to seperate Applets group.

I dont see anything else that should be changed. Please add `pr:needs-review` after I finish off that part.

File: src/citra/config.cpp (hunk 1 above)
Comment (adityaruplaha) [reply src/citra/config.cpp]: Ok, I'll start a new Applets group.

**Comment by B3n30** [comment]
@adityaruplaha @valentinvanelslande TBH both your solutions aren't good at all. We can for sure discuss which on is worse!
The only solution I think is acceptable for that is to open a QTDialog to let the user input the text and send this text to the game. (And by that I don't mean a crappy workaround with another executable) So I request to close this and stop that stupid discussion!

**Comment by mailwl** [comment]
i remember good solution - enter data from command line to swkbd (but don't remember why that PR was close)

**Comment by B3n30** [comment]
Because the console will be gone soon (see logging rewrite)

**Comment by adityaruplaha** [comment]
This is only a temporary workaround. Frankly, I don't care much about this. I personally agree with you @B3n30 but I've seen so many issues related to this, that I got irritated, & decided to go for a workaround, esp due to #3477 .

**Comment by adityaruplaha** [comment]
Thanks @B3n30 . I was just going to close this.

No sarcasm.

**Pull request closed without merging**